components/color-picker/ColorPicker.tsx

```diff
@@ -0,0 +1,170 @@
+  rootClassName?: string;
+}
+
+const ColorPicker: React.FC<ColorPickerProps> = (props) => {
+  const {
+    value,
+    defaultValue,
+    format = 'hex',
+    onFormatChange,
+    onChange,
+    allowClear = false,
+    presets,
+    children,
+    trigger = 'click',
+    open,
+    onOpenChange,
+    disabled,
+    placement = 'bottomLeft',
+    arrow = true,
+    style,
+    className,
+    rootClassName,
+    styles,
+  } = props;
+
+  const { getPrefixCls, direction } = useContext<ConfigConsumerProps>(ConfigContext);
+  const { token } = theme.useToken();
+
+  const [colorValue, setColorValue] = useColorState(token.colorPrimary, {
+    value,
+    defaultValue,
+  });
+  const [popupOpen, setPopupOpen] = useMergedState(false, {
+    value: open,
+    postState: (openData) => !disabled && openData,
+    onChange: onOpenChange,
+  });
+  const [clearColor, setClearColor] = useState(false);
+
+  const prefixCls = getPrefixCls('color-picker', customizePrefixCls);
+  const rootPrefixCls = getPrefixCls();
+
+  const [wrapSSR, hashId] = useStyle(prefixCls);
+  const mergeCls = classNames(rootClassName, className, hashId);
+  const mergePopupCls = classNames(rootClassName, hashId);
+
+  const builtinPlacements = getPlacements({
+    arrowPointAtCenter: typeof arrow === 'object' && arrow.pointAtCenter,
+    offset: token.marginXXS,
+    arrowWidth: arrow ? token.sizePopupArrow : 0,
+    borderRadius: token.borderRadius,
+  });
+
+  const currentPlacement = useMemo(() => {
+    if (!placement) {
+      return direction === 'rtl' ? 'bottomRight' : 'bottomLeft';
+    }
+
+    return placement;
+  }, [placement, direction]);
+
+  const handleChange = (data: Color) => {
+    const color: Color = generateColor(data);
+    if (clearColor && color.toHsb().a > 0) {
+      setClearColor(false);
+    }
+    if (value) {
+      onChange?.(color, color.toHexString());
+    } else {
+      setColorValue(color);
+    }
+  };
+
+  const updateColor = (colorData: Color) => {
+    handleChange(colorData);
+  };
+
+  const updateClearColor = (clear: boolean) => {
+    setClearColor(clear);
+  };
+
+  const extraProps = {
+    prefixCls,
+    builtinPlacements,
+    open,
+    trigger,
+    disabled,
+    styles,
+  };
+
+  const colorBaseProps: ColorPickerBaseProps = {
+    prefixCls,
+    color: colorValue,
+    allowClear,
+    clearColor,
+    disabled,
+    presets,
+    format,
+    onFormatChange,
+    updateColor,
+    updateClearColor,
+  };
+
+  return wrapSSR(
+    <RcColorPicker
+      value={colorValue.toHsb()}
+      classNames={{
+        popup: mergePopupCls,
+      }}
+      placement={currentPlacement}
+      arrow={!!arrow}
+      motion={{
+        motionName: getTransitionName(rootPrefixCls, 'slide-up'),
+        motionDeadline: 1000,
+      }}
+      onChange={handleChange}
+      onOpenChange={setPopupOpen}
+      panelRender={(panel) => <ExPanel {...colorBaseProps}>{panel}</ExPanel>}
+      {...extraProps}
+    >
+      {children || (
```

Comment:
```suggestion
      {children ?? (
```
我在想，这里是不是用 `??` 更合适一点，如果用户就是不想要展示 placeholder ,只想通过 open 控制显示隐藏呢？那如果用户 传一个 false 或 null 过来，依然还是会走后面的默认 placeholder

Reply:
当初设计上就需要 `trigger` 来触发，最初版本我也是这么想的，后来说了不考虑只显示 `panel`  的场景

Reply:
ok